src/Spice86.Core/Emulator/Gdb/GdbCommandBreakPointHandler.cs

```diff
@@ -141,7 +176,30 @@ private void SendS05StringToGdb() {
                 }
                 return null;
             }
-            return new AddressBreakPoint((BreakPointType)breakPointType, address, OnBreakPointReached, false);
+            
+            // Compile condition expression if present
+            Func<long, bool>? condition = null;
+            if (!string.IsNullOrWhiteSpace(conditionExpression) && _state != null && _memory != null) {
+                try {
+                    Shared.Emulator.VM.Breakpoint.Expression.ExpressionParser parser = new();
+                    Shared.Emulator.VM.Breakpoint.Expression.IExpressionNode ast = parser.Parse(conditionExpression);
+                    condition = (addr) => {
+                        BreakpointExpressionContext context = new(_state, _memory, addr);
+                        return ast.Evaluate(context) != 0;
+                    };
+                    if (_loggerService.IsEnabled(LogEventLevel.Information)) {
+                        _loggerService.Information("Compiled conditional breakpoint: {Expression}", conditionExpression);
+                    }
+                } catch (ArgumentException ex) {
+                    if (_loggerService.IsEnabled(LogEventLevel.Warning)) {
+                        _loggerService.Warning(ex, "Failed to parse condition expression: {Expression}", conditionExpression);
+                    }
+                    // Continue without condition if parsing fails
+                    conditionExpression = null;
```

Comment:
I would prefer the GDB user receives an error message. Is is handled in the GDB protocol? If so implement it